### DISCUSSION TITLE
[Bug] Skywalking + SpringCloud Gateway + Hystrix; Can not do async finish for the span repeatedly

### DISCUSSION THREAD

**Discussion by sweat123** [apache/skywalking#9646] [Ideas]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

The project will throw exception 'Can not do async finish for the span repeatedly' when the request is fallback by hystrix.
![image](https://user-images.githubusercontent.com/20635540/191198036-2b688368-a458-4329-9153-0bf180802eb0.png)

After some code debug, I find the method `asyncFinish()` will be invoked twice with the same span in `DispatcherHandlerHandleMethodInterceptor.java` due to hystrix fallback.

I have solved this in my way by adding flag in ServerWebExchange when the hystrix fallback occurred and create the span according to `ServerWebExchange` attribute flag. The test code is as follows:

![image](https://user-images.githubusercontent.com/20635540/191200618-d2e7f1a7-ee7c-43f2-b8cc-d562f2ca9a48.png)

I do not think is a good way to solve the problem. Is there a better way or can anyone give me some advice to solve this problem.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Spring gateway case is complex, and due to various use scenarios, some bugs show up from time to time.
From the SkyWalking maintainer perspective, we don't have so many Spring Gateway experiences to provide solutions or suggestions. We just made the APIs strict so you could find out about this kind of issue.

You could send a pull request for your fix to help others, and explain why you change in this way. Usually, we think fail-back and retry should be another exit span as it may call RPC again.
